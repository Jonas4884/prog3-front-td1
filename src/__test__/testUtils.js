export const todo = [{"id":1675425781448,"status":"done","value":"task 1"},{"id":1675425784921,"status":"done","value":"task 2"},{"id":1675425789408,"status":"todo","value":"task 3"},{"id":1675425831658,"status":"todo","value":"task 4"}]
export const done = [{"id":1675422371749,"status":"done","value":"clscnjsbcmks"},{"id":1675422384471,"status":"done","value":"csjcsb"},{"id":1675422586728,"status":"done","value":"cldcd"}]