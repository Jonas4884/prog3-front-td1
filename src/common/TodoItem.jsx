import { TodoStatus } from "./utils/status";


const TodoItem = (props) => {
  const { onDone, todo, onRemove } = props;

  const toDone = () => {
    onDone && onDone(todo.id);
  };

  const remove = () => {
    onRemove && onRemove(todo.id);
  };

  return (
    <div className='todo__item todo__value' role={"the__item"}>
      <input name="todo" type='checkbox' className='pointer' onClick={toDone} checked={todo.status === TodoStatus.DONE} id={todo.id.toString()} />
      <label htmlFor="todo"  className='item__value' role={"todovalue"}>{todo.value}</label>
      {todo.status === TodoStatus.DONE && (
        <p className='pointer'  role={"remove_area"} onClick={remove}>
          x
        </p>
      )}
    </div>
  );
};

export default TodoItem;
