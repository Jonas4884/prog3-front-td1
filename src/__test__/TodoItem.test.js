import { fireEvent, render, screen } from "@testing-library/react";
import renderer from 'react-test-renderer'
import TodoItem from "../common/TodoItem";
import { done, todo } from "./testUtils";

describe("should return each item of to do task", () => {
  test("should return", () => {
    render(<TodoItem  todo={todo.at(0)}/>);
    const theFirsDiv = screen.getByRole("the__item");
    expect(theFirsDiv).toBeInTheDocument();
  });

  //TO-DO : Find to test snapshot
  test("should return an chekbox by an input", () => {
    render(<TodoItem todo={todo.at(0)} />);
    const checked = screen.getByRole("checkbox")
    expect(checked).toBeInTheDocument();
  });

  test("should return a simple  checkbox", () => {
    render(<TodoItem todo={todo.at(0)}/>);
    const actual = screen.getByRole("checkbox");
    expect(actual).toBeInTheDocument();
  });

  test("should checked the checkbox", () => {
    render(<TodoItem todo={done.at(0)}/>);
    const actual = screen.getByRole("checkbox");
    expect(actual).toBeChecked();
  });

  test("should return the item value", () => {
    render(<TodoItem todo={todo.at(0)} />);
    console.log(todo.at(0).value);
    const toLabelValue = screen.findByRole("todovalue");
    console.log(toLabelValue);
    expect(toLabelValue).toBeInTheDocument();
  });

  test('should match todoitem snapshot', () => { 
    const renderComponent = renderer.create(<TodoItem todo={todo.at(0)} />).toJSON();
    expect(renderComponent).toMatchSnapshot();
   })

  test("should return a remover button to remove done task", () => {
    render(<TodoItem todo={done.at(0)} />);
    const actualStatus = screen.getByRole("remove_area");
    const remove_button = '<p class="pointer"  role="remove_area">x</p>'
    const remove_area = screen.getByText("x")
    const label =screen.getByText("clscnjsbcmks")
    expect(actualStatus).toContainHTML(
      remove_button
    );
    fireEvent.click(remove_area);
    expect(label).toBeInTheDocument();
  });

});
