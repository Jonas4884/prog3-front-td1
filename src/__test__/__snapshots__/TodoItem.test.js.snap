// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return each item of to do task should match todoitem snapshot 1`] = `
<div
  className="todo__item todo__value"
  role="the__item"
>
  <input
    checked={true}
    className="pointer"
    id="1675425781448"
    name="todo"
    onClick={[Function]}
    type="checkbox"
  />
  <label
    className="item__value"
    htmlFor="todo"
    role="todo_value"
  >
    task 1
  </label>
  <p
    className="pointer"
    onClick={[Function]}
    role="remove_area"
  >
    x
  </p>
</div>
`;
